core/src/main/scala/za/co/absa/spline/harvester/LineageHarvester.scala

```diff
@@ -142,37 +142,45 @@ class LineageHarvester(
     })
   }
 
+  private trait StackItem
```

Comment:
As these private types have no dependency on the `LigeageHarvester` instance, them had better be put in the respective companion object. Also think about moving the build creation logic into another class.